sklearn/utils/weight_vector.pyx

```diff
@@ -44,23 +42,19 @@ cdef class WeightVector(object):
         The squared norm of ``w``.
     """
 
-    def __cinit__(self,
-                  np.ndarray[double, ndim=1, mode='c'] w,
-                  np.ndarray[double, ndim=1, mode='c'] aw):
-        cdef double *wdata = <double *>w.data
+    def __cinit__(self, double [::1] w, double [::1] aw):
+        cdef double *wdata = &w[0]
 
         if w.shape[0] > INT_MAX:
             raise ValueError("More than %d features not supported; got %d."
                              % (INT_MAX, w.shape[0]))
         self.w = w
-        self.w_data_ptr = wdata
         self.wscale = 1.0
         self.n_features = w.shape[0]
         self.sq_norm = _dot(<int>w.shape[0], wdata, 1, wdata, 1)
```

Comment:
that could be replaced by `_dot(<int>w.shape[0], &w[0], 1, &w[0], 1)` to get rid of the wdata attribute